diff --git a/node_modules/next/dist/server/lib/start-server.js b/node_modules/next/dist/server/lib/start-server.js
index bec14c4..c47a13b 100755
--- a/node_modules/next/dist/server/lib/start-server.js
+++ b/node_modules/next/dist/server/lib/start-server.js
@@ -46,6 +46,14 @@ function startServer(opts) {
             resolve(app);
         });
         server.listen(port, opts.hostname);
+
+        // https://github.com/vercel/next.js/discussions/19618
+        // Ensure all inactive connections are terminated by the ALB,
+        // by setting this a few seconds higher than the ALB idle timeout
+        // https://docs.aws.amazon.com/elasticloadbalancing/latest/application/application-load-balancers.html#connection-idle-timeout
+        // Hint: the alb is pre-opening connections behind the scene
+        server.keepAliveTimeout = 185000;
+        server.headersTimeout = 190000;
     });
 }
 
